packages/mui-base/src/Option/Option.tsx

```diff
@@ -43,7 +43,7 @@ const InnerOption = React.memo(
     // If `label` is not explicitly provided, the `children` are used for convenience.
     // This is used to populate the select's trigger with the selected option's label.
     const computedLabel =
-      label ?? (typeof children === 'string' ? children : optionRef.current?.innerText);
+      label ?? (typeof children === 'string' ? children : optionRef.current?.textContent?.trim());
```

Comment:
How does this affect/fix the demo? Testing it out and getting the same value for both `innerText` and `textContent`:

https://github.com/mui/material-ui/assets/16889233/6f2e1229-446f-4b77-ac97-5afa5f5c3866

Reply:
In the "before" demo, the Select is initially "collapsed" (too short).

The difference between `innerText` and `textContent` (that I learned today) is that `innerText` takes display properties into consideration. So if an option is hidden (as it initially is), `innerText` will return ''. `textContent`, on the other hand always returns what's defined in the DOM node, regardless of visibility.

Reply:
Aah, right, it's collapsed, so it's hidden, and `innerText` doesn't return it, gotcha!